### ISSUE TITLE
Unable to sync on Mac

### ISSUE THREAD

**Issue by Batmensch** [jobisoft/TbSync#122]
## Your environment

TbSync version: 0.7.15
Thunderbird version: 60.0 (64-bit) MacOS

[X] Yes, I have installed the latest available (beta) version from 
https://github.com/jobisoft/TbSync/releases 
and my issue is not yet fixed, I can still reproduce it.

## Expected behavior
Click "Try again to connect to server", connects to server, gets calendar entries

## Actual behavior
Error: "The server does not provide information about the supported ActiveSync versions.  Is EAS blocked for this user or this client (TbSync)?  You could try to set the ActiveSync version manually."
However, ActiveSync version is manually set to v2.5

## Steps to reproduce
Fill in server config (outlook.office365.com). 

To help resolving your issue, enable debug logging (TbSync Account Manager -> Help) and send me the debug.log via e-mail (use the title of your issue as subject of the email).

**Comment by jobisoft** [comment]
can you retry with this (rename to xpi) ?
[TbSync-beta-0.7.15.5.zip](https://github.com/jobisoft/TbSync/files/2412830/TbSync-beta.zip)

**Comment by Batmensch** [comment]
Hi, that seems to work for me.  I was able to choose 2.5 from the menu options and it stuck when I ran synchronize.

**Comment by jobisoft** [comment]
Thanks, sorry for the long response time. Will release it soon.

**Comment by rduvignau** [comment]
This is an old thread but this helped me to solve a long standing synchronization problem with exchange calendars and that may help others. 

I already had issues with synchronization but lately TbSync was unable to synchronize at all my exchange calendar.

I tried different settings in TbSync options and always got "UNKNOWN_NETWORK_ERROR" in return till recreating a new account and leaving ActiveSync version on "auto" resulted into "The server does not provide information about the supported ActiveSync versions. Is EAS blocked for this user or this client (TbSync)? You could try to set the ActiveSync version manually" message. Any further tries resulted again into "UNKNOWN_NETWORK_ERROR".

Following the trick given here, I found out using my server web interface under "Options -> General -> Mobile devices" that my other synchronized devices were using ActiveSync 16.1. I then manually edited the file named  ".thunderbird/[account#].default-release/TbSync/accounts68.json", and set "asversionselected" to "16.1". After restarting thunderbird, all of a sudden my exchange calender was finally synced! I did some testing creating and deleting events, and the synchronization is working well, however, this works only if I click on "synchronize now" within the TbSync options tab not if using "synchronize" when clicking on a calendar or when clicking "Synchronize" in Thunderbird upper tab line. At last, when setting 1 minute for autosync (it was on 0), the synchronization is working all at expected. Thank you for making this work!

All in all, just something needs to be fixed here about TbSync's autodetection of ActiveSync version, I think.

**Comment by jobisoft** [comment]
TbSync does not support v16.1. It is not just about changing the version send, the protocoll has chnaged. So if you fake TbSync to "speak" v16.1, this could result in dataloss as the data exchanged with your server might be interpreted wrong. I do not suggest to do this. A server which only supports 16.1 and not an older version is very strange.

**Comment by rduvignau** [comment]
I think the server does allow all versions as I can see "allowedEasVersions":"2.0,2.1,2.5,12.0,12.1,14.0,14.1,16.0,16.1" within the json file (that has been updated once the connection was successful faking 16.1). I also think the server also understands TbSync is using 14.0 as it appears in the web interface as Mobile device details -> TbSync with client type "EAS" and protocol version "14.0"

OK, I am trying to manually set "asautoselected" to "14.0" then using config file. OK this works all right as before.

Then the problem seem to really come from the "auto" setting. Also when I was manually setting "14.0" using the graphical interface, I was always getting "UNKNOWN_NETWORK_ERROR".

**Comment by jobisoft** [comment]
Which TbSync version? Can you check the log which request to which URL failes?

**Comment by rduvignau** [comment]
I am using TbSync version 2.12.

I did not have debug mode on when I was experiencing problems and trying out the different things. OK I will try to recreate the bug, hope I will be able to get my calendar back afterwards.

OK now that I have activated debug mode, the bug is not reproducible. I left default option set to "best available" and got it working on first try this time, seems like an heisenbug...
